### PR TITLE
refactor module.file with better embedded component handling

### PR DESCRIPTION
**Pull request by erikbaranowski**
#### PR Description

This fixes up some issues with module.file running an embedded local.file component properly

#### PR Checklist

- [ ] CHANGELOG updated
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: component/module/file/file.go [hunk 1]
```diff
@@ -69,24 +72,43 @@ var (
...
 	}
-
 	if err := c.Update(args); err != nil {
 		return nil, err
 	}
 	return c, nil
 }
 
+// NewManagedLocalComponent creates the new local.file managed component.
+func (c *Component) NewManagedLocalComponent(o component.Options) (*file.Component, error) {
```
Comment (rfratto) [inline component/module/file/file.go:93]: Does this need to be exported as part of the public API of the component? Seems like we can make it internal, right?

Comment (erikbaranowski) [reply component/module/file/file.go:93]: good catch, will fix tomorrow

**Comment by erikbaranowski** [comment]
It looks like an armv6 build issue. I'm not sure what the latest status on that is.

**Comment by erikbaranowski** [comment]
merged main which removes the armv6 builds and resolved the build fail

**Review by rfratto — APPROVED** [review]
LGTM

**Pull request merged**